x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -87,10 +87,24 @@ public String executorName() {
 
     private synchronized void installDailyMaintenanceService() {
         if (mlDailyMaintenanceService == null) {
-            mlDailyMaintenanceService =
-                new MlDailyMaintenanceService(clusterService.getClusterName(), threadPool, client, clusterService, mlAssignmentNotifier);
+            mlDailyMaintenanceService = new MlDailyMaintenanceService(
+                settings,
+                clusterService.getClusterName(),
+                threadPool,
+                client,
+                clusterService,
+                mlAssignmentNotifier
+            );
             mlDailyMaintenanceService.start();
             clusterService.addLifecycleListener(new LifecycleListener() {
+                @Override
+                public void afterStart() {
+                    clusterService.getClusterSettings().addSettingsUpdateConsumer(
```

Comment:
🤔 good point. The setting updater might have to be in the initialization service itself...

To my knowledge there is no way to deregister a setting consumer. The consumers have no unique identification.

Reply:
Why is this being set to `null` anyways? I assume so it can be GC'd, but the object itself is not HUGE, and only really has references to things are referenced by the `MlInitializationService.java`. 

I am gonna change the code so that it does not get set `null`. It seems like a waste to me. 

@dimitris-athanasiou @droberts195 ^ Let me know if you have a prevailing opinion the other way.